src/json.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NINJA_JSON_H_
+#define NINJA_JSON_H_
+
+#include <stdio.h>
+
+void EncodeJSONString(FILE* stream, const char* str) {
```

Comment:
Maybe `std::ostream&` instead of `FILE*` so that we can use a `std::stringstream` instead of `tmpfile()`?

Reply:
In that case, shall we limit the use of `std::cout` to `printCompdb` only, or propagate it to ToolCompilationDatabase to be consistent? std::ios_base::sync_with_stdio by default is on and tends to slow down stuff a lot.

Or make it `std::string EncodeJSONString(const std::string& str)`? We're already calling c_str() in `printCompdb` anyway

(I [originally](https://github.com/ninja-build/ninja/commit/be0daf4870635ec5076b2caff500237a3b2519e8#diff-6d51fafa9848c43417816f023be8ee23ceabf5963620c740aadb0c8d1930580bR33) used `open_memstream` to avoid `tmpfile` but  `open_memstream` isn't available on Windows)

Reply:
I just did some `time ninja -t compdb` experiments with build.ninja generated from chromium (ran only once):
* [master](https://github.com/ninja-build/ninja/commit/c8f8f2a9e3016ab7a9ecb2e8b084cf441f3ae88e) 7.50 sec
* `void EncodeJSONString(FILE* stream, const char* str)` 7.30 sec; [code](https://github.com/ninja-build/ninja/commit/450f977cd0a0c240fde3fa6d58a10b5ac0342801)
* `std::string EncodeJSONString(const std::string& str)` with fwrite 6.86 sec; [code](https://github.com/afq984/ninja/commit/82a27729ede4184cabfcf8fd3465659a3c64b0d2)
* `EncodeJSONString(std::ostream& os, const std::string& in)` with std::stringstream::str().c_str() and fwrite 11.39sec; [code](https://github.com/afq984/ninja/commit/9e4bf10300b61b5562e75e297a6c8f9de3d4c6bc)
* `void EncodeJSONString(std::ostream& os, const std::string& in) with std::cout` 13.94 sec; [code](https://github.com/afq984/ninja/commit/d1d5be20e0e685bdc9cad92f3c03d10dcd95e786)

Reply:
That's a bit sad :( Thanks for testing.

Reply:
@jhasse I updated the PR to use `std::string EncodeJSONString(const std::string& str)`, please take a look!